core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -185,25 +189,32 @@ public void start() {
    }
 
    protected void pollStoreAvailability() {
-      storesMutex.writeLock().lock();
+      storesMutex.readLock().lock();
       try {
-         boolean availabilityChanged = false;
-         boolean failureDetected = false;
-         for (StoreStatus status : storeStatuses.values()) {
-            if (status.availabilityChanged())
-               availabilityChanged = true;
-            if (availabilityChanged && !status.availability && !failureDetected) {
-               failureDetected = true;
-               unavailableException = new StoreUnavailableException(String.format("Store %s is unavailable", status.store));
-               cacheNotifier.notifyPersistenceAvailabilityChanged(false);
+         // If the lock is not available, then it means that an availability check is in progress due to scheduled calls overlapping
```

Comment:
The method can't be invoked multiple times from the scheduler per the javadoc, it just gets delayed. The only way this could be concurrently called is if someone disabled a store at the same time.